Plugin/Sales/Model/Spi/InvoiceResource.php

```diff
@@ -194,12 +166,14 @@ public function aroundLoad(
         /** @var \Magento\Framework\Model\ResourceModel\Db\AbstractDb $resultEntity */
         $resultEntity = $proceed($entity, $value, $field);
 
-        // Load AvaTax extension attributes
+        // Load AvaTax record into extension attributes
         if ($this->avaTaxConfig->isModuleEnabled($entity->getStoreId())) {
 
-            // Get the AvaTax Attributes from the AbstractModel
-            $avataxIsUnbalanced = $entity->getData('avatax_is_unbalanced');
-            $baseAvataxTaxAmount = $entity->getData('base_avatax_tax_amount');
+            // Get the AvaTax record
+            /** @var Invoice $invoice */
+            $avataxRecord = $this->avataxInvoice->loadByParentId($entity->getId());
```

Comment:
Eliminate the loadByParentId() method on your model and resourceModel by using the Abstract load() method with a field name parameter as indicated on the CreditMemo comment.

Reference: \Magento\Framework\Model\ResourceModel\Db\AbstractDb::load